src/overridable.test.js

```diff
@@ -0,0 +1,208 @@
+import {mount} from 'enzyme';
+import PropTypes from 'prop-types';
+import React, {Component} from 'react';
+import Overridable, {parametrize, OverridableContext} from '.'; // export from index.js to test it
+
+class ExampleComponent extends Component {
+  static propTypes = {
+    title: PropTypes.string.isRequired,
+    color: PropTypes.oneOf(['blue', 'green', 'red']),
+    children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
+  };
+
+  static defaultProps = {
+    color: 'blue',
+    children: null,
+  };
+
+  render() {
+    const {title, color, children} = this.props;
+    return (
+      <Overridable id="ExampleComponent.container" cmpTitle={title} cmpColor={color}>
+        <>
+          <div style={{color}}>{title}</div>
+          {children}
+        </>
+      </Overridable>
+    );
+  }
+}
+
+const OverridableExampleComponent = Overridable.component('ExampleComponent', ExampleComponent);
+
+const assertTitleStyle = (element, title, style) => {
+  expect(
+    element.filterWhere(el => {
+      const styleAttr = el.prop('style');
+      // simple obj comparison for equality
+      const sameStyle = JSON.stringify(styleAttr) === JSON.stringify(style);
+      return el.prop('children') === title && sameStyle;
+    })
+  ).toHaveLength(1);
+};
+
+describe('Tests for default cmp behaviour', () => {
+  test('it should render the default cmp with default props and passed children', () => {
+    const Cmp = mount(
+      <OverridableExampleComponent title="TODOs">
+        <p>
+          <span>First TODO</span>
+          <span>Second TODO</span>
+        </p>
+      </OverridableExampleComponent>
+    );
+
+    assertTitleStyle(Cmp.find('div'), 'TODOs', {color: 'blue'});
+
+    const children = Cmp.find('p').children();
+    expect(children).toHaveLength(2);
+    expect(children.get(0)).toEqual(<span>First TODO</span>);
+    expect(children.get(1)).toEqual(<span>Second TODO</span>);
+  });
+
+  test('it should render the default component with passed props and no children', () => {
+    const Cmp = mount(<OverridableExampleComponent title="TODOs" color="green" />);
+
+    assertTitleStyle(Cmp.find('div'), 'TODOs', {color: 'green'});
+
+    const children = Cmp.find('p').children();
+    expect(children).toHaveLength(0);
+  });
+});
+
+describe('Tests for parametrized', () => {
+  /**
+   * With using parametrize, the final component will be rendered:
+   *
+   * <Overridable(ExampleComponent) title="TODOs">
+   *  <Parametrized(Overridable(ExampleComponent)) title="TODOs">
+   *    <ExampleComponent title="New title" color="red">
+   */
+  test('it should render the cmp with id `ExampleComponent` with new props', () => {
+    const parametrized = parametrize(OverridableExampleComponent, {
+      title: 'New title',
+      color: 'red',
+    });
+    const overriddenCmps = {ExampleComponent: parametrized};
```

Comment:
I know that it works like this, but to be consistent with the example on README, what about using an explicit string here?
`const overriddenCmps = {'ExampleComponent': parametrized};`

Reply:
It is ESLint + Prettier that is removing it :)

Reply:
I'd rather remove the quotes from the docs ;)

Reply:
Oh, then we can do the other way round, modify [this](https://github.com/indico/react-overridable/pull/1/files#diff-04c6e90faac2675aa89e2176d2eec7d8R58)

Reply:
Changed in the doc!